src/apps/components/juttle-editor.js

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+
+import AceEditor from 'juttle-react-ace';
+
+import { promulgateBundle } from '../actions';
+import { LOCAL_BUNDLE_ID } from '../constants';
+
+import 'brace/theme/monokai';
+import './juttle-mode/mode';
+
+const FONT_SIZE = 14;
+const EDIT_EVAL_INTERVAL_MS = 500;
+const SCROLL_MARGIN_TOP = 15;
+const SCROLL_MARGIN_BOTTOM = 15;
+
+class JuttleEditor extends React.Component {
+    onChange = (newValue) => {
+        if (!this.props.readOnly) {
+            this._updateProgram(newValue);
+            localStorage.setItem('program', newValue);
```

Comment:
suggestion: In the spirit of trying to keep as much logic out of the views, perhaps it might be better for the redux update action to set localStorage